src/measure_theory/borel_hierarchy.lean

```diff
@@ -0,0 +1,505 @@
+    use λ n, (⟨f n, typf n⟩ : ⋃ (j < order.succ (sup o)), pi0 s j),
+    tauto },
+  { simp only [Union_subset_iff],
+    exact λ _ hi, sigma0_subset_sigma0 s _ _ (le_of_lt hi) }
+end
+
+theorem compl_mem_gen_measurable (t : set α) (ht : t ∈ gen_measurable s) :
+  tᶜ ∈ gen_measurable s :=
+begin
+  rw gen_measurable_eq_Union_sigma0 at ht,
+  simp only [mem_Union,exists_prop] at ht,
+  cases ht with o ho,
+  rcases classical.em (o=0) with rfl | onon,
+  { finish },
+  calc
+  tᶜ  ∈ pi0 s o          : by { rw pi0_eq_compl_sigma0,
+    simp only [mem_image,compl_inj_iff,exists_eq_right], exacts [ho.2,onon] }
+  ... ⊆ gen_measurable s : pi0_subset_sigma0 s o ω₁ ho.1,
+end
+
+theorem Union_mem_gen_measurable {f : ℕ → set α} (hf : ∀ n, f n ∈ gen_measurable s) :
+  (⋃ n, f n) ∈ gen_measurable s:=
+by { unfold gen_measurable at *, exact Union_of_mem_sigma0 s ω₁ hf }
+
+open measurable_space
+
+lemma generate_measurable_of_mem_sigma0 (t) (ht : t ∈ sigma0 s i) :
+  generate_measurable s t :=
+begin
+  induction i using ordinal.induction with i IH generalizing t,
+  rw [sigma0_eq_Union_pi0,mem_range] at ht,
+  rcases ht with ⟨f,hf⟩,
+  have typf : ∀ n : ℕ, generate_measurable s (f n),
+  { intro n,
+    have fn_in : (f n).val ∈ ⋃ (j : ordinal) (hij : j < i), pi0 s j := (f n).property,
+    simp only [subtype.val_eq_coe,mem_Union,exists_prop] at fn_in,
+    rcases fn_in with ⟨o,⟨o_lt_i,fn_in⟩⟩,
+    -- Case `(f n).val ∈ pi0 s 0`.
+    rcases classical.em (o=0) with rfl | honz,
+    { rw pi0_zero at fn_in,
+      rcases fn_in with  fn_in | fn_emp | fn_in,
+      { exact generate_measurable.basic _ fn_in },
+      { rw fn_emp, exact generate_measurable.empty },
+      { rw mem_singleton_iff at fn_in, rw [fn_in,←compl_empty],
+        exact generate_measurable.compl _ generate_measurable.empty } },
+    -- Case `(f n).val ∈ pi0 s o` with `o ≠ 0`.
+    simp only at IH,
+    rw pi0_eq_compl_sigma0 s o honz at fn_in,
+    rw ← compl_compl ↑(f n),
+    apply generate_measurable.compl,
+    cases fn_in with x hx,
+    rw [←hx.2, compl_compl],
+    exact IH o o_lt_i x hx.1 },
+  rw ← hf,
+  exact generate_measurable.union (λn, (f n).val) typf
+end
+
+theorem generate_measurable_eq_gen_measurable :
+  {t | generate_measurable s t} = gen_measurable s :=
+begin
+  ext t, refine ⟨λ ht, _, λ ht, _⟩,
+  { have om1_nonz : ω₁ ≠ 0,
+    { unfold omega_1, exact ne_zero_of_out_nonempty _,},
+    induction ht with u hu u hu IH f hf IH,
+    exacts
+    [ self_subset_sigma0 s ω₁ om1_nonz hu,
+      empty_mem_sigma0 s ω₁ om1_nonz,
+      compl_mem_gen_measurable s u IH,
+      Union_mem_gen_measurable s IH ] },
+  { exact generate_measurable_of_mem_sigma0 s ω₁ t ht }
+end
+
+end gen_measurable
+
+section card_gen_measurable
+
+/-!
+### Cardinality of sigma-algebras
+
+This section includes the same results (with essentially the same proofs) from
+`measure_theory.card_measurable_space` by Gouëzel.
```

Comment:
You can remove this, and also delete the file `measure_theory.card_measurable_space`. I think it would make sense to put this section and the removal in another PR, to keep a manageable size for the current PR (PRs of more than 400 lines are often much harder to get in than shorter ones, because reviewer time is harder to find for such PRs).

Reply:
Thanks for the insight. I was confused on how PRs worked and actually [put more than I intended](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2317972.20Borel.20hierarchy) into in this one.

So, to be clear what the next steps are, I will:
- delete the section on cardinalities of sigma-algebras;
- wait for this PR to be merged;
- make a new branch (or reuse this one) to add a new file containing the aforementioned section;
- delete `measure_theory.card_measurable_space` in the new branch;
- make a new PR for these changes.

Alternatively: is it reasonable to replace the contents of `measure_theory.card_measurable_space` with my corresponding formalized results?

Another question: The new file will be really similar to the deleted one. Should I list you and Violeta among the authors?

Reply:
Yes, the next steps look perfect. You can definitely replace the contents of `measure_theory.card_measurable_space` with the results from your formalization, and keep myself and Violeta as authors. That's probably the best way to proceed.